Sources/Widgets/Widgets3D/LineWidget/index.js

```diff
@@ -60,29 +71,32 @@ function vtkLineWidget(publicAPI, model) {
     HandleRepresentationType.CIRCLE
   ] = vtkCircleHandleRepresentation;
 
-  function initializeHandleRepresentations() {
-    HandleRepresentation[0] = shapeToRepresentation[model.handle1Shape];
-    if (!HandleRepresentation[0]) {
-      HandleRepresentation[0] = vtkSphereHandleRepresentation;
-    }
-    HandleRepresentation[1] = shapeToRepresentation[model.handle2Shape];
-    if (!HandleRepresentation[1]) {
-      HandleRepresentation[1] = vtkSphereHandleRepresentation;
-    }
-  }
-
   model.methodsToLink = [
     'activeScaleFactor',
     'activeColor',
     'useActiveColor',
     'glyphResolution',
     'defaultScale',
   ];
-  model.behavior = widgetBehavior;
-  model.widgetState = stateGenerator();
-  model.widgetState.setPositionOnLine(model.positionOnLine);
-  model.widgetState.setNbHandles(0);
-  initializeHandleRepresentations();
+
+  publicAPI.initializeHandleRepresentations = () => {
+    model.handle1Shape = model.widgetState.getHandle1().getShapeName();
+    model.handle2Shape = model.widgetState.getHandle2().getShapeName();
```

Comment:
I do not think this is how mix-ins are supposed to be handled.
To me, it shoud be handled inside "ArrowHandleRepresentation::requestData"

Reply:
about this review, my understanding of the problem is that requestData should have a glyph to display. In order to do so it should know on initialization which is the glyph to display with requestData. How is it possible to request a data not set before?

Reply:
Maybe `requestData` should call `detectArrowShape`

Reply:
sorry to discuss the review like that but the only solution i came up with is not really elegant. RequestData is looped at each event, like a mouseMove when nothing is selected at all (eg initial state of the scene before the user grabs focus). So to apply detectArrowShape in requestData would mean to set a glyph at null on initialization, then catch it with a bool variable to detect when is the 1st requestData sent. To me it is overkill. Am i missing something here?

Reply:
The idea is to call `detectArrowShape` in `requestData` (or in a function calling requestData) only when the "shape" changes. 
I guess it is fine to keep a variable to know what is the current shape set as input of mapper.
I think it is elegant because you can change the shape without having to reinstantiate the class.